packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -138,8 +138,17 @@ const SigninContainer = ({
     localizedErrorMessage: localizedErrorFromLocationState,
   } = location.state || {};
 
+  const email = queryParamModel.email || emailFromLocationState;
+
+  const { sessionToken, uid } = getAccountInfo(email);
+
   const [needsCanLinkAccountWebChannel, setNeedsCanLinkAccountWebChannel] =
-    useState(integration.isSync() && location.pathname.includes('force_auth'));
+    useState(
+      integration.isSync() &&
+        location.pathname.includes('force_auth') &&
+        sessionToken &&
+        email !== queryParamModel.email
```

Comment:
Should this be compared to the email in local storage/currently signed in account?